scrapy/contrib/loader/processor.py

```diff
@@ -70,3 +74,254 @@ def __init__(self, separator=u' '):
+                if value != '' and value is not None:
+                    filtered.append(value)
+            return filtered
+        if self.except_chars:
+            return filter(lambda v: v.isalpha() or (v in self.except_chars), values)
+        else:
+            return filter(type(values).isalpha, values)
+
+
+class OnlyCharsItems(object):
+
+    def __init__(self, except_chars=''):
+        self.except_chars = except_chars
+
+    def __call__(self, values):
+        if isinstance(values, list):
+            filtered = []
+            for value in values:
+                condition = value.isalpha()
+                if not condition and self.except_chars:
+                    condition = purge_chars(value, self.except_chars).isalpha()
+                if value != '' and value is not None and condition:
+                    filtered.append(value)
+            return filtered
+        else:
+            condition = values.isalpha()
+            if not condition:
+                condition = purge_chars(values, self.except_chars).isalpha()
+            if values != '' and values is not None and condition:
+                return values
+
+
+class OnlyDigits(object):
+
+    def __init__(self, except_chars=''):
+        self.except_chars = except_chars
+
+    def __call__(self, values):
+        if isinstance(values, list):
+            filtered = []
+            for value in values:
+                if self.except_chars:
+                    value = filter(lambda v: v.isdigit() or (v in self.except_chars), value)
+                else:
+                    value = filter(type(value).isdigit, value)
+                if value != '' and value is not None:
+                    filtered.append(value)
+            return filtered
+        if self.except_chars:
+            return filter(lambda v: v.isdigit() or (v in self.except_chars), values)
+        else:
+            return filter(type(values).isdigit, values)
+
+
+class OnlyDigitsItems(object):
+
+    def __init__(self, except_chars=''):
+        self.except_chars = except_chars
+
+    def __call__(self, values):
+        if isinstance(values, list):
+            filtered = []
+            for value in values:
+                condition = value.isdigit()
+                if not condition and self.except_chars:
+                    condition = purge_chars(value, self.except_chars).isdigit()
+                if value != '' and value is not None and condition:
+                    filtered.append(value)
+            return filtered
+        else:
+            condition = values.isdigit()
+            if not condition:
+                condition = purge_chars(values, self.except_chars).isdigit()
+            if values != '' and values is not None and condition:
+                return values
+
+
+class Filter(object):
+
+    def __init__(self, filter_func):
+        self.filter_func = filter_func
+
+    def __call__(self, values):
+        if isinstance(values, list):
```

Comment:
Do not differentiate between list and value. Filter what it comes in the call. If you want to iterate and filter the elements of a list individually you can always use `MapCompose`. otherwise you are forcing to wrap values in a list.